app/controllers/api_controller/providers.rb

```diff
@@ -5,7 +5,7 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port security_protocol).freeze
```

Comment:
we support both types with /api/providers, see provider_class="provider" parameter. This capability was added when Foreman was being introduced.